app/models/pseud.rb

```diff
@@ -497,4 +491,15 @@ def self.search(options={})
     end
   end
 
+  def should_reindex_works?
+    pertinent_attributes = %w[id name]
+    destroyed? || (saved_changes.keys & pertinent_attributes).present?
+  end
+
+  # If the pseud gets renamed, anything with the old name needs to be reindexed.
+  def reindex_works
+    return unless should_reindex_works?
+    IndexQueue.enqueue_ids(Work, works.pluck(:id), :main)
+    IndexQueue.enqueue_ids(Bookmark, bookmarks.pluck(:id), :main)
```

Comment:
I'm having a brain issue -- is this reindexing the pseud's bookmarks, or bookmarks of works (and series?) by the pseud? Either way, if it's indexing more than just works, maybe a better name would be `reindex_creations`?

Reply:
It's indexing works by the pseuds and bookmarks by the pseuds. I can rename the method.

This hook is not indexing bookmarks of works (and series) by the pseud (or bookmarkables by the pseud). Previously this wasn't a problem because when we updated the work (with `set_author_sorting`), the bookmarkable callbacks will run, but they no longer do. It's a new bug. :<

When user name changes, I'll add tests to check that these get reindexed:

- works by affected pseuds
- bookmarkables by affected pseuds (only works and series; external works have no pseuds)
- bookmarks by affected pseuds (since pseud names are indexed as bookmarkers)

Reply:
When we reindex a work, its bookmarkable index gets updated as well, so I didn't add any bug there.

I did miss series reindexing though.

Reply:
I love this new world order \o/

This looks good to go now 👍